test/adaptive_load/fake_plugins/fake_metrics_plugin/fake_metrics_plugin_test.cc

```diff
@@ -0,0 +1,148 @@
+#include "adaptive_load/plugin_loader.h"
+#include "api/adaptive_load/benchmark_result.pb.h"
+#include "api/client/options.pb.h"
+#include "envoy/registry/registry.h"
+#include "test/adaptive_load/fake_plugins/fake_metrics_plugin/fake_metrics_plugin.h"
+
+#include "external/envoy/source/common/config/utility.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+namespace Nighthawk {
+namespace {
+
+using ::nighthawk::adaptive_load::BenchmarkResult;
+using ::nighthawk::adaptive_load::FakeMetricsPluginConfig;
+using ::nighthawk::adaptive_load::MetricEvaluation;
+using ::nighthawk::client::CommandLineOptions;
+using ::testing::HasSubstr;
+
+TEST(FakeMetricsPluginConfigFactory, CreateEmptyConfigProtoCreatesCorrectType) {
+  auto& config_factory =
+      Envoy::Config::Utility::getAndCheckFactoryByName<MetricsPluginConfigFactory>(
+          "nighthawk.fake-metrics-plugin");
+  Envoy::ProtobufTypes::MessagePtr empty_config = config_factory.createEmptyConfigProto();
+  FakeMetricsPluginConfig expected_config;
+  EXPECT_EQ(empty_config->DebugString(), expected_config.DebugString());
+  EXPECT_TRUE(Envoy::MessageUtil()(*empty_config, expected_config));
+}
+
+TEST(FakeMetricsPluginConfigFactory, FactoryRegistersUnderCorrectName) {
```

Comment:
Now that I think about it, isn't it impossible for this to fail, if the plugin got into the registry under the name? @oschaaf I remember when we added this -- do you think it's safe to delete?

Reply:
Well, it explicitly adds coverage of the implementation of `::name()` -- and naively I don't know what complex voodoo could hypothetically be responsible for returning an instantiation with the right name. So this test just proves that `::name()` doesn't crash and returns what is expected, and will ensure that stays put as we update Envoy / make changes to the involved implementations.

If coverage is realised "by proxy" via another test, that probably suffices, and this test might be superfluous. We're testing test code here after all? But ... thinking more about it, maybe more value would be added if this would be done as a `TYPED_TEST` that iterates over all implementations that could/should be found/discovered in the registry? To get a feel for the lines along which I'm thinking here, see `TYPED_TEST(TypedStatisticTest, createNewInstanceOfSameType)` over at https://github.com/envoyproxy/nighthawk/blob/dc5cd242a522aab034488a73ab60c28c41a20def/test/statistic_test.cc#L102

Reply:
Let's keep this test for now (there will also be similar ones in other PRs), but I filed #471 to explore TYPED_TEST which might obsolete all such tests.

Reply:
sgtm, thanks @eric846 for considering & filing that issue.